internal/log/log.go

```diff
@@ -67,8 +47,17 @@ func entry(ctx context.Context) *logrus.Entry {
 	id, idOk := ctx.Value(ID{}).(string)
 	name, nameOk := ctx.Value(Name{}).(string)
 	if idOk && nameOk {
-		return logger.WithField("id", id).WithField("name", name)
+		return logger.WithField("id", id).WithField("name", name).WithContext(ctx)
 	}
 
-	return logrus.NewEntry(logger)
+	return logrus.NewEntry(logger).WithContext(ctx)
+}
+
+func StartSpan(ctx context.Context) (context.Context, trace.Span) {
+	spanName := ""
+	// Use function signature as a span name if available
+	if pc, _, _, ok := runtime.Caller(1); ok {
+		spanName = runtime.FuncForPC(pc).Name()
+	}
+	return trace.SpanFromContext(ctx).TracerProvider().Tracer("").Start(ctx, spanName)
```

Comment:
```suggestion
	return trace.SpanFromContext(ctx).TracerProvider().Tracer("crio").Start(ctx, spanName)
```

Reply:
my understanding was that using `""` would get us default (== `crio`) so that we wouldn't have to hardcode it twice

Reply:
Ah alright.